base-notebook/test/test_container_options.py

```diff
@@ -196,24 +225,75 @@ def test_sudo_path_without_grant(container):
     rv = c.wait(timeout=10)
     assert rv == 0 or rv["StatusCode"] == 0
     logs = c.logs(stdout=True).decode("utf-8")
+    assert "ERROR" not in logs
+    assert "WARNING" not in logs
     assert logs.rstrip().endswith("/opt/conda/bin/jupyter")
 
 
-def test_group_add(container, tmpdir):
+def test_group_add(container):
     """Container should run with the specified uid, gid, and secondary
-    group.
+    group. It won't be possible to modify /etc/passwd since gid is nonzero, so
+    additionally verify that setting gid=0 is suggested in a warning.
     """
     c = container.run(
         user="1010:1010",
-        group_add=["users"],
+        group_add=["users"],  # Ensures write access to /home/jovyan
         command=["start.sh", "id"],
     )
     rv = c.wait(timeout=5)
     assert rv == 0 or rv["StatusCode"] == 0
     logs = c.logs(stdout=True).decode("utf-8")
+    assert "ERROR" not in logs
+    warnings = [
+        warning for warning in logs.split("\n") if warning.startswith("WARNING")
+    ]
+    assert len(warnings) == 1
+    assert "Try setting gid=0" in warnings[0]
```

Comment:
I don't understand, could you elaborate?

Reply:
The way you check that you have some kind of expected warning.

Few lines above you do: `assert warnings[0].startswith("WARNING: Jupyter Notebook deprecation notice")`
Here you do: `assert "Try setting gid=0" in warnings[0]`

I think first option is better, because it is more concrete.

Reply:
I'm already checking that the line begins with "WARNING" and I just wanted to verify that I'm making the suggestion.

The full message is
```bash
_log "WARNING: unable to fix missing /etc/passwd entry because we don't have write permission. Try setting gid=0 with \"--user=$(id -u):0\"."
```

Would you really prefer that I use `startswith` here? I already explain in the test docstring what's going on, so it doesn't seem to me like it improves anything.

Reply:
Ok, I think you're right.